wpimath/src/main/java/edu/wpi/first/math/controller/PIDController.java

```diff
@@ -327,6 +327,7 @@ public double getVelocityError() {
    */
   public double calculate(double measurement, double setpoint) {
     m_setpoint = setpoint;
+    m_haveSetpoint = true;
```

Comment:
why not use `PIDController::setSetpoint(double)` instead?

Reply:
Because that function also updates the error values which is wasted work since calculate() also does that later.